aspnet/fundamentals/routing.rst

```diff
@@ -1,10 +1,273 @@
-.. include:: /../common/stub-topic.txt
+Routing
+=======
+By `Steve Smith`_
 
-.. _routing-index:
+Routing middleware is used to map requests to route handlers. Routes are configured when the application starts up, and can define values that will be passed as parameters to route handlers. Routing functionality is also responsible generating links that correspond to routes in ASP.NET apps.
 
-|stub-icon| Routing
-===================
+Sections:
+	- `Routing Middleware`_
+	- `Configuring Routing`_
+	- `Template Routes`_
+	- `Link Generation`_
+
+`View sample files <https://github.com/aspnet/Docs/tree/1.0.0-rc1/aspnet/fundamentals/routing/sample>`_
+
+Routing Middleware
+------------------
+The routing :doc:`middleware` uses *routes* to map requests to an ``IRouteHandler``. The ``IRouteHandler`` returns a ``RequestDelegate``, or ``null``. If it returns a delegate, that delegate is invoked; otherwise, routing continues. If no route handler is found for a request, then the middleware calls *next* (and the next middleware in the request pipeline is invoked).
+
+.. TODO in RC2 Middleware later in the request pipeline can access route data using extensions methods on ``HttpContext``.
+
+To get started using routing in your app, add it to the **dependencies** in ``project.json``:
+
+.. literalinclude:: routing/sample/RoutingSample/project.json
+  :linenos:
+  :dedent: 2
+  :language: javascript
+  :lines: 6-12
+  :emphasize-lines: 4
+  
+Next, add it to ``ConfigureServices`` in ``Startup.cs``:
+
+.. literalinclude:: routing/sample/RoutingSample/Startup.cs
+  :linenos:
+  :dedent: 8
+  :language: c#
+  :lines: 16-19
+  :emphasize-lines: 3
+
+With this in place, you will have access to routing in your app. The next step is to configure it.
+
+Configuring Routing
+-------------------
+Routing is configured in the ``Configure`` method in your ``Startup`` class. Create an instance of `RouteBuilder <https://github.com/aspnet/Routing/blob/1.0.0-rc1/src/Microsoft.AspNet.Routing/RouteBuilder.cs>`_. You can optionally set the ``ServiceProvider`` and/or ``DefaultHandler`` properties, in order to make them available as you build routes.
```

Comment:
consider removing 
Create an instance of `RouteBuilder <https://github.com/aspnet/Routing/blob/1.0.0-rc1/src/Microsoft.AspNet.Routing/RouteBuilder.cs>`_.
Keeping it has the advantage of the link provided, however it would be better to link to the API doc.